packages/tools/kadena-cli/src/account/utils/addAccount.ts

```diff
@@ -0,0 +1,29 @@
+import type { CommandResult } from '../../utils/command.util.js';
+import type { IAddAccountManualConfig } from '../types.js';
+import { getFilePath, getUpdatedConfig } from './addHelpers.js';
+import { validateAccountDetails } from './validateAccountDetails.js';
+import { writeConfigInFile } from './writeConfigInFile.js';
+
+export async function addAccount(
+  config: IAddAccountManualConfig,
+  callback: () => Promise<boolean>,
```

Comment:
Isn't this missing something? When I define a callback to decide between local and on-chain data, it would help to actually have access to the local and on-chain data to compare.